### PR TITLE
Redesign Execute Page UI

### LINKED ISSUE #169: [F] Update condition selection component
Update the condition select screen to make it more user friendly

- Include some form of user hints to help them understand what the conditions mean
  - Tooltips
  - Modal with large writeup of instructions (can be separate issue)

- Design changes to be determined by @alex-hobby 

# Milestones
| Milestone                	| Date   	|
|--------------------------	|--------	|
| Determine styling changes  | Oct 28    |
| Rough draft of interface 	| Nov 2   	|
| Styling changes             	| Nov 4	|
| Finalized                          	| Nov 6	|

### PR DESCRIPTION
**Pull request by alex-hobby**
## Redesign Execute Page UI

# Description

Completely redesigned Execute Page

Features:
- static sidebar: contains connection info, database switcher, procedure switcher, and execute area
- main content: divided into three tabs (rule groups, unit tests, results)
- Rule groups / unit tests: "add" button to add, each added quantity will appear in a black box (scrolls if overflowing). expandable quantities are in accordions
- Delete entries with trash icon
- Add Rule Group Form: name, procedure in disabled input, procedure parameters
- Add Unit Test Form: name, table -> select level -> fill out remaining conditional fields
- Error checking in both forms

Code Niceties:
- custom hook for modals (useModal) which provides isOpen / toggle
- helper function to create connection string

TO BE DONE:
- Add unit tests to rule groups
- Edit rule groups/unit tests that have been added (stretch goal)
- Remove conditionList variable from execute as only rules should know about their unit tests
- Save created tests / rules to backend
- Fix logic for executing - it's still very much hard coded

## Link to the issue you are closing

Fixes #169 

## Screenshot or other testing that you have completed

https://user-images.githubusercontent.com/29695042/199619732-04cd036c-4e45-4470-9704-9945d0f13612.mp4

# Checklist:

- [x] ESLint passes
- [x] Prettier passes
- [x] Included comments where necessary
- [x] Updated README if needed
- [x] Verified that code runs and generates no errors/warnings

### CODE REVIEW

**Comment by insaan21** [comment]
Looks good to me, maybe we could change test level to be test type instead, I think it might make more sense

File: src/renderer/components/execute/conditions/UnitTestTab.tsx [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  const { isOpen, toggle } = useModal();
+
+  const deleteCondition = (condition: Partial<UnitTestType>) => {
+    const newList = conditionList.filter(
+      (c: Partial<UnitTestType>) => c !== condition
+    );
+    setConditionList(newList);
+  };
+
+  const addCondition = (condition: Partial<UnitTestType>) => {
```
Comment (insaan21) [inline src/renderer/components/execute/conditions/UnitTestTab.tsx:22]: If you're adding directly to the conditionList, do you need to call setConditionList()

**Review by insaan21 — APPROVED** [review]
lgtm apart from two nit's I had

File: src/renderer/components/execute/conditions/UnitTestTab.tsx (hunk 1 above)
Comment (JackFratto) [reply src/renderer/components/execute/conditions/UnitTestTab.tsx:22]: Yes, setConditionList is a hook which when called React checks if the prop values changed and if so re-renders the component

File: src/renderer/components/execute/ParameterContainer.tsx [hunk 2]
```diff
@@ -1,9 +1,13 @@
 import { Form, InputGroup } from 'react-bootstrap';
 import { ProcedureParameter } from 'db/Procedures';
+import { useEffect, useState } from 'react';
+import { IRuleGroupErrors } from './rules/AddRuleGroupModal';
 
+/* eslint-disable @typescript-eslint/no-explicit-any */
 interface IParameterContainerProps {
   activeParameters: ProcedureParameter[];
-  handleInput: (inputValue: string, attribute: string) => void;
+  setField: (f: string, v: any) => void;
```
Comment (JackFratto) [inline src/renderer/components/execute/ParameterContainer.tsx:9]: not sure if f is field and v are value, could we make this more explicit?

File: src/renderer/components/execute/ParameterContainer.tsx [hunk 3]
```diff
@@ -12,28 +16,60 @@ export type Parameter = {
...
+    });
+  };
+
+  useEffect(() => {
+    const defaultValues: Parameter = {};
+    activeParameters.forEach((item) => {
+      defaultValues[item.name] = '';
+    });
+    setParameterValues(defaultValues);
+  }, [activeParameters]);
```
Comment (JackFratto) [inline src/renderer/components/execute/ParameterContainer.tsx:37]: shouldn't this be parameterValues?

Comment (JackFratto) [reply src/renderer/components/execute/ParameterContainer.tsx:37]: I'm kind of assuming that's what the second useEffect is for, but isn't it possible to combine the two? I'm not really sure what the one below this does

Comment (alex-hobby) [reply src/renderer/components/execute/ParameterContainer.tsx:37]: This useeffect updates the parameter values state when the active parameters change. The other one fires when parameter values changes, which happens in two places so it needs its own use effect. You're right that this useeffect will fire that other one as well 100% of the time

**Pull request merged**